examples/notebooks/FastaiLRFinder_MNIST.ipynb

```diff
@@ -218,18 +218,53 @@
     "from matplotlib import pyplot as plt\n",
     "ax = lr_finder.plot()\n",
     "plt.axes(ax)\n",
-    "plt.show()"
+    "plt.show()\n",
+    "\n",
+    "print(\"Suggested LR\", lr_finder.lr_suggestion())"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {
     "id": "EHHNoOi8mvd6"
    },
+   "source": [
+    "Let's compare training results: \n",
```

Comment:
I think to convey a better message on how to use the LR finder we have to change the order of below runs.
The idea is to show how we can use LR finder in user code without re-creating model and optimizer again.
So, let's continue the training with suggested LR and create new model / optimizer with initial params.

By the way, the line "Note, that model and optimizer were restored to their initial states when FastaiLRFinder finished." should be put here and not below.

Reply:
> So, let's continue the training with suggested LR and create new model / optimizer with initial params.

Can you please explain this more?

Reply:
Think about how would you use this feature in your project:
```python
model = ...
optimizer = ...
trainer = ...

# do lr finder trick
# ...

lr_finder.apply_suggested_lr(optimizer)

trainer.run(...)
```

We do not want to create another model etc and this should be shown here as an example.

Reply:
I tried this approach already, but I found nothing on how to reset the parameters I found this discussion https://discuss.pytorch.org/t/reset-optimizer-stats/78516/5?u=kickitlikeshika , https://github.com/pytorch/pytorch/issues/37410

Reply:
Why would you like to reset parameters in this case ?

Reply:
Should we continue training with same old parameters? that will be easier to implement, but I though we needed to show the user that a new model/optimizer with found LR will perform way better, that's why I wanted to start from scratch again..

Reply:
Let me explain again my point. The notebook is nice now and almost OK, but it does not show the user how to use the feature in their project as we are creating new models/optim etc. In the doc I suggest to write it like that : 
```python
model = ...
optimizer = ...

init_model_state = model.state_dict()
init_opt_state = optimizer.state_dict()

# ----

trainer = create_supervised_trainer(model, optimizer, criterion, device=device)
ProgressBar(persist=True).attach(trainer, output_transform=lambda x: {"batch loss": x})

lr_finder = FastaiLRFinder()
to_save={'model': model, 'optimizer': optimizer}
with lr_finder.attach(trainer, to_save, diverge_th=1.5) as trainer_with_lr_finder:
    trainer_with_lr_finder.run(trainloader)

# ----

from matplotlib import pyplot as plt
ax = lr_finder.plot()
plt.show()

print("Suggested LR", lr_finder.lr_suggestion())

# ----

Let's compare training results:

    a) with applying optimal learning rate.
    b) using a fixed learning rate (lr=3e-4)

# ----

lr_finder.apply_suggested_lr(optimizer)
print(optimizer.param_groups[0]['lr'])

# ----

trainer.run(trainloader, max_epochs=10)

evaluator = create_supervised_evaluator(model, metrics={"acc": Accuracy(), "loss": Loss(nn.NLLLoss())}, device=device)
evaluator.run(testloader)

print(evaluator.state.metrics)

# ----

# Reinit model / optim
model.load_state_dict(init_model_state)
optimizer.load_state_dict(init_opt_state)

# ----

trainer.run(trainloader, max_epochs=10)

evaluator = create_supervised_evaluator(model, metrics={"acc": Accuracy(), "loss": Loss(nn.NLLLoss())}, device=device)
evaluator.run(testloader)

print(evaluator.state.metrics)

```